app/models/manageiq/providers/kubevirt/infra_manager/metrics_capture/prometheus_capture_context.rb

```diff
@@ -0,0 +1,73 @@
+class ManageIQ::Providers::Kubevirt::InfraManager::MetricsCapture::PrometheusCaptureContext < ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::PrometheusCaptureContext
+  def initialize(target, start_time, end_time, interval)
+    @target = target
+    @starts = start_time.to_i.in_milliseconds
+    @ends = end_time.to_i.in_milliseconds if end_time
+    @interval = interval.to_i
+    @tenant = target.try(:container_project).try(:name) || '_system'
+    @ext_management_system = @target.ext_management_system
+    @ts_values = Hash.new { |h, k| h[k] = {} }
+    @metrics = []
+
+    validate_target
+  end
+
+  def prometheus_endpoint
+    @ext_management_system.parent_manager.connection_configurations.prometheus.endpoint
+  end
+
+  def prometheus_credentials
+    {:token => @ext_management_system.parent_manager.authentication_token("prometheus")}
+  end
+
+  def prometheus_options
+    {
+      :http_proxy_uri => VMDB::Util.http_proxy_uri.to_s.presence,
+      :verify_ssl     => @ext_management_system.parent_manager.verify_ssl_mode(prometheus_endpoint),
+      :ssl_cert_store => @ext_management_system.parent_manager.ssl_cert_store(prometheus_endpoint),
+      :open_timeout   => 5,
+      :timeout        => 30
+    }
+  end
+
+  def collect_metrics
+    labels = labels_to_s(:name => @target.name)
+
+    collect_metrics_for_labels(labels)
+  end
+
+  def collect_metrics_for_labels(labels)
+    # prometheus field is in core usage per sec
+    # miq field is in pct of node cpu
+    #
+    #   rate is the "usage per sec" readings avg over last 5m
+
+    cpu_resid = "sum(rate(kubevirt_vmi_cpu_user_usage_seconds_total{#{labels}}[#{AVG_OVER}]))"
+    # fetch_counters_data(cpu_resid, 'cpu_usage_rate_average', @node_cores / 100.0)
+    fetch_counters_data(cpu_resid, 'cpu_usage_rate_average', 1)
+
+    # prometheus field is in bytes, @node_memory is in mb
+    # miq field is in pct of node memory
+    mem_resid = "sum(kubevirt_vmi_memory_domain_bytes{#{labels}})"
+    # fetch_counters_data(mem_resid, 'mem_usage_absolute_average', @node_memory * 1e6 / 100.0)
```

Comment:
```suggestion
```

Reply:
yeah I plan on removing those, kept them in for this comment https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/260#discussion_r1854655603